### PR TITLE
Make a basic User model (including secure passwords)

### PR DESCRIPTION
**Pull request by NguyenDinhDuy**

### CODE REVIEW

File: app/assets/javascripts/users.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (dat-hedspi) [inline app/assets/javascripts/users.coffee]: remove this file

File: app/assets/stylesheets/users.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (dat-hedspi) [inline app/assets/stylesheets/users.scss]: remove this file

File: Gemfile [hunk 3]
```diff
@@ -1,6 +1,7 @@
 source "https://rubygems.org"
 
 gem "rails", "~> 5.0.0", ">= 5.0.0.1"
+gem "bcrypt",         "3.1.11"
```
Comment (nguyenhuutien) [inline Gemfile]: Sau , 1 dấu cách thôi

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save {self.email = email.downcase}
+  validates :name, presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+                    format: {with: VALID_EMAIL_REGEX},
```
Comment (dat-hedspi) [inline app/models/user.rb]: ident tương tự dưới

File: config/routes.rb [hunk 5]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'users/new'
```
Comment (dat-hedspi) [inline config/routes.rb]: double quote

File: test/controllers/users_controller_test.rb [hunk 6]
```diff
@@ -1,9 +1,9 @@
-require "test_helper"
+require 'test_helper'
```
Comment (dat-hedspi) [inline test/controllers/users_controller_test.rb]: double quote

**Pull request merged**